test/foundry/ReentrancyGuard.t.sol

```diff
@@ -79,11 +79,7 @@ contract ReentrancyCaller {
     }
 }
 
-abstract contract ReentrancyGuardErrors {
-    error ReentrancyFail();
-}
-
-contract ReentrancyGuardTest is TestHelpers, ReentrancyGuardErrors {
+contract ReentrancyGuardTest is TestHelpers {
```

Comment:
Is this going to inherit from `IReetrancyGuard`?

Reply:
Actually, it reverts on the fallback and I could not catch the reversion with `vm.expectRevert()` (although I see, as expected, the error `ReentrancyFail` in the stack traces). That's why I didn't import the interface.

https://github.com/LooksRare/contracts-libs/blob/165c6e9babf6ed27cafef7d038bdf29ae350706d/test/foundry/ReentrancyGuard.t.sol#L106

Reply:
Try this in SafeFaucet, you have an empty `revert()` so the `ReentrancyFail` is not bubbling up, failing with `EvmError: revert` instead

```
assembly {
    status := call(gas(), to, amount, 0, 0, 0, 0)
    // returndatacopy(t, f, s)
    // copy s bytes from returndata at position f to mem at position t
    returndatacopy(0, 0, returndatasize())
    switch status
    case 0 {
        // revert(p, s)
        // end execution, revert state changes, return data mem[p…(p+s))
        revert(0, returndatasize())
    }
}
```

Reply:
Nice, it worked well. Let me adjust.